app/controllers/targets_controller.rb

```diff
@@ -42,7 +42,7 @@ def mark_as_read
         .where(courses: { id: course.id })
         .first if current_user.present?
 
-    student.page_reads.where(target: @target).first_or_create
+    student.page_reads.create_or_find_by(target: @target)
```

Comment:
This will not work as intended. `create_or_find_by` is incompatible with Rails' model-level uniqueness validation, and so this will fail with a `RecordInvalid` on duplicate calls.

This part of the change is probably okay - I think the line `validates :student_id, uniqueness: { scope: :target_id }` needs to be removed from the table.

As far as I can tell, there is no case where a user will ever need to be shown a validation failure that a page read already exists.

Reply:
Got it. Removed the model validation.
This was working when I tested because there was no `!` and hence it was failing silently for the duplicated page read.